src/engine/specification.md

```diff
@@ -155,35 +156,60 @@ This structure contains the attributes required to initiate a payload build proc
-3. Client software **MUST** return `{status: SYNCING, latestValidHash: null}` if the sync process is already in progress or if requisite data for payload validation is missing. In the event that requisite data to validate the payload is missing (e.g. does not have payload identified by `parentHash`), the client software **SHOULD** initiate the sync process.
+1. Client software **SHOULD** validate the payload if requisite data for payload validation is locally available. The validation process is specified in the [Payload validation process](#payload-validation-process) section. Additionally:
+    * Client software **MUST** return the result of the payload validation if the validation has been initiated.
 
-4. Client software **MAY** provide additional details on the validation error if the payload is deemed `INVALID` by assigning the corresponding message to the `validationError` field.
+1. Client software **MUST** return `{status: ACCEPTED, latestValidHash: null, validationError: null}` if neither payload validation nor sync process has been initiated.
 
-5. Client software **MUST** return `-32002: Invalid terminal block` error if the parent block is a PoW block that doesn't satisfy terminal block conditions according to [EIP-3675](https://eips.ethereum.org/EIPS/eip-3675#definitions). This check maps on the Transition block validity section of [EIP-3675](https://eips.ethereum.org/EIPS/eip-3675#transition-block-validity).
+1. If any of the above fails due to errors unrelated to the normal processing flow of the method, the client software **MUST** return an error.
+
+##### Payload validation process
+
+Payload validation process consists of validating a payload with respect to the block header and execution environment rule sets. The process is specified as follows:
+
+1. Client software **MUST** validate the payload according to the block header and execution environment rule set with modifications to these rule sets defined in the [Block Validity](https://eips.ethereum.org/EIPS/eip-3675#block-validity) section of [EIP-3675](https://eips.ethereum.org/EIPS/eip-3675#specification). Additionally:
+    * If validation succeeds, the response **MUST** contain `{status: VALID, latestValidHash: payload.blockHash}`
+    * If validation fails, the response **MUST** contain `{status: INVALID, latestValidHash: validHash}` where `validHash` is the block hash of the most recent *valid* ancestor of the invalid payload. That is, the valid ancestor of the payload with the highest `blockNumber`
+    * Client software **MUST** discard the payload if it's deemed invalid.
+
+1. Client software **MUST** return `-32002: Invalid terminal block` error if the parent block is locally available and is a PoW block that doesn't satisfy terminal block conditions according to [EIP-3675](https://eips.ethereum.org/EIPS/eip-3675#definitions). This check maps on the Transition block validity section of [EIP-3675](https://eips.ethereum.org/EIPS/eip-3675#transition-block-validity).
```

Comment:
I was thinking about this edge case. It seems like a client software can't respond with error because it cannot validate `A` with respect to the terminal block conditions if `A` is missing. In this case it should rather respond with `INVALID`, but then the question what should it return as `latestValidHash`. It should be `latestValidHash: parentOfA.block_hash`, but CL won't be able to do anything with this value as there is no such block in its database. Probably, CL may look at the `parent_hash` of `A` in this case to conclude that a blockchain starting at `A` and ending at `B` is invalid.

I would replace this error with `INVALID` but need to decide is returned as `latestValidHash`.